chevah/compat/tests/normal/test_capabilities.py

```diff
@@ -218,17 +251,121 @@ def test_isPrivilegeEnabled_disabled(self):
         privilege = win32security.SE_SECURITY_NAME
         self.assertFalse(self.capabilities._isPrivilegeEnabled(privilege))
 
-    def test_isPrivilegeEnabled_absent(self):
+    def test_symbolic_link(self):
         """
-        Returns False for a privilege which is not present.
+        Not supported on Windows without elevated permissions.
         """
-        # By default SE_RELABEL_NAME should not be available to test
-        # accounts.
-        privilege = win32security.SE_RELABEL_NAME
-        self.assertEqual(
-            u'absent', self.capabilities._getPrivilegeState(privilege))
+        symbolic_link = self.capabilities.symbolic_link
 
-        self.assertFalse(self.capabilities._isPrivilegeEnabled(privilege))
+        self.assertFalse(symbolic_link)
+
+    def test_impersonate_local_account_windows(self):
+        """
+        Impersonation is not available when running as a normal user.
+        """
+        result = self.capabilities.impersonate_local_account
+
+        self.assertFalse(result)
+
+    def test_create_home_folder(self):
+        """
+        On Windows home folders can be created if required privileges
+        are configured for the process.
+        """
+        result = self.capabilities.create_home_folder
+
+        if 'win-xp' in self.hostname:
+            self.assertFalse(result)
+        else:
+            self.assertTrue(result)
+
+    def test_getCurrentPrivilegesDescription(self):
+        """
+        Check getCurrentPrivilegesDescription.
+        """
+        text = self.capabilities.getCurrentPrivilegesDescription()
+
+        self.assertContains('SeChangeNotifyPrivilege:3', text)
+
+        if 'win-xp' in self.hostname:
+            self.assertNotContains('SeCreateSymbolicLinkPrivilege', text)
+            self.assertNotContains('SeImpersonatePrivilege', text)
+            # Windows XP has SeCreateGlobalPrivilege enabled.
+            self.assertContains('SeCreateGlobalPrivilege:3', text)
+        else:
+            self.assertNotContains('SeCreateSymbolicLinkPrivilege', text)
+            self.assertNotContains('SeImpersonatePrivilege', text)
+            self.assertNotContains('SeCreateGlobalPrivilege', text)
+
+
+class TestNTProcessCapabilitiesAdministrator(CompatTestCase):
+    """
+    Capability tests executed only on Windows slaves that are configured to
+    run with administrator rights.
+    """
+
+    def setUp(self):
+        super(TestNTProcessCapabilitiesAdministrator, self).setUp()
+
+        if self.os_family != 'nt':
+            raise self.skipTest("Only Windows platforms supported.")
+
+        # Windows 2008 and DC client tests are done in administration mode,
+        # 2003 and XP under normal mode.
+        if 'win-2003' in self.hostname or 'win-xp' in self.hostname:
+            message = "Only Windows with administrator rights supported"
+            raise self.skipTest(message)
```

Comment:
this whole dance is ok, but it would be nice to have it as an skipOnCondition decorator.

the main reason as that with this code we will still execute the setUp code for each test, which if we have a skipOnCondition for the whole testcase the setUp code will never be executed